internal/common/util/list.go

```diff
@@ -45,3 +45,11 @@ func Filter[T any](list []T, predicate func(val T) bool) []T {
 	}
 	return out
 }
+
+func Map[T any, U any](list []T, fn func(val T) U) []U {
```

Comment:
this is a nit but I think really this should be in a package called `slices` so its `slices.Map()` (as opposed to `arrays.Map()` or `maps.Map()`).